### ISSUE TITLE
[FEATURE]  Maria Greer: antecendent losses

### ISSUE THREAD

**Issue by chasegan** [chasegan/Kalix#146]
Maria wants a loss node that can switch between antecedent and normal conditions based on the last few months of rainfall. 

Hard switches are kinda yuck. I wonder if we can just do something smooth. Maybe we just use a normal loss node (with a table that represents the most antecedent version of the loss) but add a function that can modulate the loss (0 - 100%). fn.antecedent_percentage = 100 * min(1, max(0, 130/six_month_rain))

Update:
- Need "rate = " on loss node.
- This should be an absolute value (not a proportion!) because absolute rates e.g. 8.112 ML/d are the natural output of loss tables. Working with proportions is significantly harder across a range of examples. 
- We use "rate = " instead of "loss = " so that the modeller can use values that exceed the actual flow e.g. "rate = 1e9" without causing an issue. The recorded "rate" value will be 1d9, but the recorded "loss" value will be the actual loss based on the flows.

**Comment by chasegan** [comment]
Ask Maria if she has a case with data that I can use to test performance of options.  

[edit] 
Email sent

**Comment by chasegan** [comment]
Chas notes:

Paul's recalibration folder didn't contain any sort of documentation explaining he was looking at, but I suspect I have found the bits in the Lower Balonne model he was doing funky wet/dry stuff:

There are 4 reaches with 100GL virtual storages used to track antecedent conditions. These fill up in floods and evaporate slowly. The states of these storages are used to split river flows between a wet-period loss table, and dry-period loss table. -- We can replace those dummy storage reaches with simple variable expressions in Kalix (and avoid the dummy network structures). The functions are simple and much cleaner than the dummy reaches. Kalix losses currently only have a tabular loss, but if I include a way to set the loss via an expression, then we can easily roll back and forth between two tables based on the antecedent condition tracker var. This should be more flexible and clearer.

At Narran Lake there is an 8GL storage to do initial losses. -- This is probably covered by the reach dead storage feature that I'm currently putting in. (Something NSW use often).
